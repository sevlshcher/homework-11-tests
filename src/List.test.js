// Компонент List - список
// Напишите для него тесты.
// Компонент позволяет добавлять элементы в список.
// При добавлении нового элемента тексовое поле очищается.

// При клике на элементы списка они удаляются.

// Убедитесь, что вы протестировали всю функциональнось.

import React from 'react'
import { List } from './List'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

describe('List', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<List />).toJSON()
        expect(tree).toMatchSnapshot( )
    })

    describe('working correctlly', () => {
        const wrapper = shallow(<List />);
        const item = 'someItem';
    
        it('add item', () => {
          const input = wrapper.find('.input');
    
          input.simulate('change', { target: { value: item } });
          wrapper.find('.addButton').simulate('click');
          expect(wrapper.find('.list').contains(item)).toEqual(true);
        });
    
        it('clear input', () => {
          const input = wrapper.find('.input');
          const value = input.prop('value');
    
          expect(value).toEqual('');
        });
    
        it('remove item', () => {
          wrapper.find('.removeLi').first().simulate('click');
          expect(wrapper.find('.removeLi').length).toEqual(0);
        })
      })
})