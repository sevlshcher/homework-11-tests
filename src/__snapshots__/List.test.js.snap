// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
Array [
  <ul
    className="list"
  />,
  <input
    className="input"
    onChange={[Function]}
    value=""
  />,
  <button
    className="addButton"
    onClick={[Function]}
  >
    Add item
  </button>,
]
`;
